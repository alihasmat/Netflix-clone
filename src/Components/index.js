export { default as Accordian } from "./Accordian/Accordian";
export { default as Card } from "./Card/Card";
export { default as Feature } from "./Feature/Feature";
export { default as Footer } from "./Footer/Footer";
export { default as Form } from "./Form/Form";
export { default as Header } from "./Header/Header";
export { default as Jumbotron } from "./Jumbotron/Jumbotron";
export { default as Loading } from "./Loading/Loading";
export { default as Player } from "./Player/Player";
export { default as Profiles } from "./Profiles/Profiles";
export { default as Search } from "./Search/Search";
