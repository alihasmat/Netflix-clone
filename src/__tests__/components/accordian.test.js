import { render, fireEvent } from "@testing-library/react";
import { Accordian } from "../../Components";
import faqsData from "../../fixtures/faqs.json";

describe("<Accordian />", () => {
  it(`renders <Accordian /> with populated data.`, () => {
    const { container, getByText } = render(
      <Accordian>
        <Accordian.Title>Frequently Asked Questions</Accordian.Title>
        {faqsData.map((item) => (
          <Accordian.Item key={item.id}>
            <Accordian.Header>{item.header}</Accordian.Header>
            <Accordian.Body>{item.body}</Accordian.Body>
          </Accordian.Item>
        ))}
      </Accordian>
    );
    expect(getByText("Frequently Asked Questions")).toBeTruthy();
    expect(getByText("What is Netflix?")).toBeTruthy();
    expect(getByText("How much does Netflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Netflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it(`opens and closes <Accordian /> component`, () => {
    const { container, queryByText } = render(
      <Accordian>
        <Accordian.Title>Frequently Asked Questions</Accordian.Title>
        {faqsData.map((item) => (
          <Accordian.Item key={item.id}>
            <Accordian.Header>{item.header}</Accordian.Header>
            <Accordian.Body>{item.body}</Accordian.Body>
          </Accordian.Item>
        ))}
      </Accordian>
    );
    const whatIsNetflixTextBody =
      "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

    expect(queryByText(whatIsNetflixTextBody)).toBeFalsy();
    fireEvent.click(queryByText("What is Netflix?"));
    expect(queryByText(whatIsNetflixTextBody)).toBeTruthy();

    fireEvent.click(queryByText("What is Netflix?"));
    expect(queryByText(whatIsNetflixTextBody)).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
