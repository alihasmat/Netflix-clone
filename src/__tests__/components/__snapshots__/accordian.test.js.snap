// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordian /> opens and closes <Accordian /> component 1`] = `
<div
  class="style__Container-sc-1ieakz8-0 ccmevn"
>
  <div
    class="style__Inner-sc-1ieakz8-1 ccebQc"
  >
    <h1
      class="style__Title-sc-1ieakz8-2 bsnaC"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordian /> renders <Accordian /> with populated data. 1`] = `
<div
  class="style__Container-sc-1ieakz8-0 ccmevn"
>
  <div
    class="style__Inner-sc-1ieakz8-1 ccebQc"
  >
    <h1
      class="style__Title-sc-1ieakz8-2 bsnaC"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        What is Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        How much does Netflix cost?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        Where can I watch?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        How do I cancel?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="style__Item-sc-1ieakz8-3 hrogMD"
    >
      <div
        class="style__Header-sc-1ieakz8-4 ivreYx"
      >
        What can I watch on Netflix?
        <img
          alt="open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
